Realm/Tests/MigrationTests.mm

```diff
@@ -307,6 +307,25 @@ - (void)testStringPrimaryKeyNoIndexMigration {
     [RLMRealm migrateRealmAtPath:RLMTestRealmPath()];
 }
 
+- (void)testIntPrimaryKeyNoIndexMigration {
+    // make string an int
+    RLMObjectSchema *objectSchema = [RLMObjectSchema schemaForObjectClass:MigrationPrimaryKeyObject.class];
+
+    // create without search index
+    objectSchema.primaryKeyProperty.attributes = 0;
```

Comment:
Ah, that's right. well that's the best we can do until we move away from `RLMPropertyAttributes` altogether (#935).